src/transformers/models/speecht5/modeling_speecht5.py

```diff
@@ -2539,7 +2539,7 @@ def _generate_speech(
     vocoder: Optional[nn.Module] = None,
     output_cross_attentions: bool = False,
 ) -> Union[torch.FloatTensor, Tuple[torch.FloatTensor, torch.FloatTensor]]:
-    encoder_attention_mask = torch.ones_like(input_values)
+    encoder_attention_mask = (1 - (input_values==model.config.pad_token_id).int())
```

Comment:
Would it be possible to update the signature of `.generate` to accept an attention mask, and then forward this to the model? See https://github.com/huggingface/transformers/issues/25908#issuecomment-1706963362 for details

Reply:
Sure, added in the latest commit.